blocks/library/image/index.js

```diff
@@ -70,10 +70,10 @@ export const settings = {
 			{
 				type: 'raw',
 				isMatch( node ) {
-					const tag = node.nodeName.toLowerCase();
-					const hasImage = node.querySelector( 'img' );
+					const isImage = node.nodeName === 'IMG';
+					const hasImage = !! node.querySelector( 'img' );
 
-					return tag === 'img' || ( hasImage && tag === 'figure' );
+					return isImage || hasImage;
```

Comment:
We are accepting any node, provided it contains an image, not certain if there are cases where something with an image tag inside should not be transformed into another block but I think right now that is not case and we are safe.